internal/controller/cluster_create.go

```diff
@@ -302,45 +302,100 @@ func (r *ClusterReconciler) reconcilePoolerSecrets(ctx context.Context, cluster
 }
 
 func (r *ClusterReconciler) reconcilePostgresServices(ctx context.Context, cluster *apiv1.Cluster) error {
-	if configuration.Current.CreateAnyService {
-		anyService := specs.CreateClusterAnyService(*cluster)
-		cluster.SetInheritedDataAndOwnership(&anyService.ObjectMeta)
+	anyService := specs.CreateClusterAnyService(*cluster)
+	cluster.SetInheritedDataAndOwnership(&anyService.ObjectMeta)
 
-		if err := r.serviceReconciler(ctx, anyService); err != nil {
-			return err
-		}
+	if err := r.serviceReconciler(ctx, anyService, configuration.Current.CreateAnyService, cluster); err != nil {
+		return err
 	}
 
 	readService := specs.CreateClusterReadService(*cluster)
 	cluster.SetInheritedDataAndOwnership(&readService.ObjectMeta)
 
-	if err := r.serviceReconciler(ctx, readService); err != nil {
+	if err := r.serviceReconciler(ctx, readService, cluster.IsReadServiceEnabled(), cluster); err != nil {
 		return err
 	}
 
 	readOnlyService := specs.CreateClusterReadOnlyService(*cluster)
 	cluster.SetInheritedDataAndOwnership(&readOnlyService.ObjectMeta)
 
-	if err := r.serviceReconciler(ctx, readOnlyService); err != nil {
+	if err := r.serviceReconciler(ctx, readOnlyService, cluster.IsReadOnlyServiceEnabled(), cluster); err != nil {
 		return err
 	}
 
 	readWriteService := specs.CreateClusterReadWriteService(*cluster)
 	cluster.SetInheritedDataAndOwnership(&readWriteService.ObjectMeta)
 
-	return r.serviceReconciler(ctx, readWriteService)
+	if err := r.serviceReconciler(ctx, readWriteService, cluster.IsReadWriteServiceEnabled(), cluster); err != nil {
+		return err
+	}
+
+	return r.reconcileManagedServices(ctx, cluster)
 }
 
-func (r *ClusterReconciler) serviceReconciler(ctx context.Context, proposed *corev1.Service) error {
+func (r *ClusterReconciler) reconcileManagedServices(ctx context.Context, cluster *apiv1.Cluster) error {
```

Comment:
A default service could also be specified inside the managed services list. This would cause it to be reconciled with the template after the initial creation. 
This could help migrate the default services to the managed template.
I am looking forward to others' opinions on the matter.

Reply:
I agree with Armando's proposal to rely on a default to facilitate the migration to the managed services.